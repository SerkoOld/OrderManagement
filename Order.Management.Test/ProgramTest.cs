using System;
using System.IO;
using System.Reflection;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Xunit;

namespace Order.Management.Test
{
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("Approvals")]
    public class ProgramTest
    {
        [Fact]
        public void TestMainOutputs()
        {
            using var fakeStdout = new StringWriter();
            Console.SetOut(fakeStdout);

            var currentDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
            using var inputNames = new StreamReader($@"{currentDir}/Approvals/input.txt");
            // Tell console to get its input from the file, not from the keyboard
            Console.SetIn(inputNames);

            Program.Main(new string[] { });
            Approvals.Verify(fakeStdout.ToString());
        }
    }
}