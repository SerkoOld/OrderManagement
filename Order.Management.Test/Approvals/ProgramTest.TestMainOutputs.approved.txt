Please input your Name: Please input your Address: Please input your Due Date: 
Please input the number of Red Square: 
Please input the number of Blue Square: 
Please input the number of Yellow Square: 
Please input the number of Red Triangle: 
Please input the number of Blue Triangle: 
Please input the number of Yellow Triangle: 
Please input the number of Red Circle: 
Please input the number of Blue Circle: 
Please input the number of Yellow Circle: 
Your invoice report has been generated: 

Name: David Wei Address: 26A Croydon Street, Sydenham, Christchurch 8024 Due Date: 29-09-2021 Order #: 0
-------------------------------------------------------------------------
|                 |       Red       |      Blue       |     Yellow      |
-------------------------------------------------------------------------
|     Square      |        1        |        2        |        3        |
|    Triangle     |        4        |        7        |        2        |
|     Circle      |        5        |        8        |        1        |
-------------------------------------------------------------------------

Squares 		  6 @ $1 ppi = $6
Triangles 		  13 @ $2 ppi = $26
Circles 		  14 @ $3 ppi = $42
Red Color Surcharge       10 @ $1 ppi = $10

Your cutting list has been generated: 

Name: David Wei Address: 26A Croydon Street, Sydenham, Christchurch 8024 Due Date: 29-09-2021 Order #: 0
--------------------
|         |   Qty   |
--------------------
| Square  |    6    |
|Triangle |   13    |
| Circle  |   14    |
--------------------

Your painting report has been generated: 

Name: David Wei Address: 26A Croydon Street, Sydenham, Christchurch 8024 Due Date: 29-09-2021 Order #: 0
-------------------------------------------------------------------------
|                 |       Red       |      Blue       |     Yellow      |
-------------------------------------------------------------------------
|     Square      |        1        |        2        |        3        |
|    Triangle     |        4        |        7        |        2        |
|     Circle      |        5        |        8        |        1        |
-------------------------------------------------------------------------
